### DISCUSSION TITLE
Select picker on API Rest

### DISCUSSION THREAD

**Discussion by jaquelinedealmeida** [Budibase/budibase#13749] [Help] [answered]
Hello. I'm trying to do the following:
I have the following fields:
1- State that uses the data provider with the localities/states query
2- City that uses the data provider with a query {{UF}}/cities

UF is an acronym for states.

When the user selects the state, only cities corresponding to the chosen state must be returned in the city field. 

I have already passed some conditions, filters and rules, but I was not successful.
How should I proceed?

I used binding UF on query city, but I don't know if is important
![Captura de tela 2024-05-22 172129](https://github.com/Budibase/budibase/assets/79230534/361d6d81-95ad-4e4f-ae00-bfe32a57a6d4)
![Captura de tela 2024-05-22 172156](https://github.com/Budibase/budibase/assets/79230534/0a9a8d42-e974-4e38-be59-40116ecdeaf1)
![Captura de tela 2024-05-22 172422](https://github.com/Budibase/budibase/assets/79230534/c2a37acb-00f1-44d7-bb50-3853b39dae33)
![Captura de tela 2024-05-22 214541](https://github.com/Budibase/budibase/assets/79230534/fda2ace3-cd01-44a5-9aa1-083e4ad559f9)

I want to do as this https://github.com/Budibase/budibase/discussions/3983

**Comment by mikesealey** [comment, marked as answer]
Hi @jaquelinedealmeida 

I've been looking into this for you. I'll give you an overview of how I approached this. I've also attached an app-export for you to take a look at. 
[13749-export-1716465337141.tar.gz](https://github.com/Budibase/budibase/files/15416401/13749-export-1716465337141.tar.gz)

I've set two REST API queries using a GET method, the first getting States/Estados
![image](https://github.com/Budibase/budibase/assets/39269380/eec86486-bff8-4865-be5b-5de503646cfe)

and the second gets Municipalities/Municipios (forgive me, my Portuguese is non-existant).
![image](https://github.com/Budibase/budibase/assets/39269380/9080e3c9-57ff-496f-9652-96839b4eae08)

Notice the Binding added to the second query, where the binding name appears inside handlebars of the query path. You can also set a default here if no value is specified. This can be useful for debugging.

.

In the design tab, my component tree looks like this
![image](https://github.com/Budibase/budibase/assets/39269380/13327287-b53f-4f06-87f6-d4169343b905)

Where the first State Data Provider simply returns all states, and feeds them into the State options picker. (You should take care to set the Limit in the Data Provider high enough to return all options)

The Municipality Data Provider is set slightly differently, it includes on extra step. Clicking the sprocket/cog next to the Data setting will open up a bindings menu
![image](https://github.com/Budibase/budibase/assets/39269380/2200e72c-6b20-49a3-8545-82b27ca2e9cd)

.

In the bindings menu you will need to specify a value for UF in the URL of your Municipality/Municipios query. This allows that value to be passed into the data tab, so it knows which state you're specifying when running the query. I'm setting it with the value of the above field using handlebar notation.
![image](https://github.com/Budibase/budibase/assets/39269380/109cda23-a997-4145-aea4-16fb0601ae88)

The result is that the Municipality/Municipios options picker shows options dependant on the State options picker.

As your Municipality/Municipios picker depends on States/Estados, you should set Municipality/Municipios to disabled, and in the Conditions tab on the right you can add a condition to update the disabled setting to unchecked (enabled) if the State/Estados field is not empty. This will help keep your users on track and ensure you get good quality data from your form.

![image](https://github.com/Budibase/budibase/assets/39269380/3e34e544-39f8-4175-9aec-bdb0cb977023)



Hope this helps. If you have any more questions please let us know!

**Comment by jaquelinedealmeida** [comment]
Thank you!! After thinking so much I had got to resolve it! I followed the same way that you!!

**Comment by jaquelinedealmeida** [comment]
I followed the same step to district!
